device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsTransport.java

```diff
@@ -55,6 +55,9 @@
     /** Messages whose callbacks that are waiting to be invoked. */
     private final Queue<IotHubCallbackPacket> callbackList = new LinkedBlockingDeque<>();
 
+    /** Connection state change callbacks */
+    private final List<IotHubConnectionStateCallback> stateCallbackList = new ArrayList<>();
+
```

Comment:
For my particular project we would only ever have a single callback registered.  I have no issues changes this to be a member variable and not a List.